### PR TITLE
DO_NOT_MERGE: feature/prl-4259_changes_to_generic_screen

### PR DESCRIPTION
**Pull request by JamesCutts00**
Changes to hering need reverting, hardcoded in for testing purposes

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PRL-4259

### Change description ###

Generic screen changes for create an order from Swansea

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/DraftAnOrderController.java [hunk 1]
```diff
@@ -104,6 +105,18 @@ public AboutToStartOrSubmitCallbackResponse populateFl404Fields(
...
+            CaseData caseData = objectMapper.convertValue(
+                    callbackRequest.getCaseDetails().getData(),
+                    CaseData.class
+            );
+            if (caseData.getManageOrders() != null && caseData.getManageOrders()
+                    .getJudgeOrMagistrateTitle() == JudgeOrMagistrateTitleEnum
+                    .justicesLegalAdviser && (caseData.getJusticeLegalAdviserFullName() == null || caseData
+                    .getJusticeLegalAdviserFullName().isBlank())) {
+                List<String> errorList = new ArrayList<>();
+                errorList.add("Full name of Justices' Legal Advisor is mandatory, when the Judge's title is selected as Justices' Legal Adviser");
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/DraftAnOrderController.java:117]: Move this error message to constant

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java [hunk 2]
```diff
@@ -192,6 +194,18 @@ public AboutToStartOrSubmitCallbackResponse populateJudgeOrAdminDraftOrderCustom
                 callbackRequest.getCaseDetails().getData(),
                 CaseData.class
             );
+            List<String> errorList = new ArrayList<>();
+            if (caseData.getManageOrders() != null && caseData.getManageOrders()
+                    .getJudgeOrMagistrateTitle() == JudgeOrMagistrateTitleEnum
+                    .justicesLegalAdviser && (caseData.getJusticeLegalAdviserFullName() == null || caseData
+                    .getJusticeLegalAdviserFullName().isBlank())) {
+                errorList.add("Full name of Justices' Legal Advisor is mandatory, when the Judge's title is selected as Justices' Legal Adviser");
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java:202]: All error mesage needs to go to constants

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java [hunk 3]
```diff
@@ -226,6 +240,17 @@ public AboutToStartOrSubmitCallbackResponse populateCommonFields(
                 CaseData.class
             );
             Map<String, Object> response = draftAnOrderService.populateCommonDraftOrderFields(authorisation, caseData);
+            CreateSelectOrderOptionsEnum selectedOrder = (CreateSelectOrderOptionsEnum) response.get("orderType");
+            List<String> errorList = new ArrayList<>();
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java:244]: ErrorList can be created once beginning of the method and can be used wherever needed.

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java [hunk 4]
```diff
@@ -192,6 +194,18 @@ public AboutToStartOrSubmitCallbackResponse populateJudgeOrAdminDraftOrderCustom
...
             );
+            List<String> errorList = new ArrayList<>();
+            if (caseData.getManageOrders() != null && caseData.getManageOrders()
+                    .getJudgeOrMagistrateTitle() == JudgeOrMagistrateTitleEnum
+                    .justicesLegalAdviser && (caseData.getJusticeLegalAdviserFullName() == null || caseData
+                    .getJusticeLegalAdviserFullName().isBlank())) {
+                errorList.add("Full name of Justices' Legal Advisor is mandatory, when the Judge's title is selected as Justices' Legal Adviser");
+                return AboutToStartOrSubmitCallbackResponse.builder().errors(errorList).build();
+            }
+            if (caseData.getDateOrderMade() == null || caseData.getDateOrderMade().toString().isBlank()) {
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java:205]: This condition should be refactored

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/ManageOrdersController.java [hunk 5]
```diff
@@ -138,6 +143,19 @@ public AboutToStartOrSubmitCallbackResponse populatePreviewOrderWhenOrderUploade
...
+            if (caseData.getManageOrders() != null && caseData.getManageOrders()
+                    .getJudgeOrMagistrateTitle() == JudgeOrMagistrateTitleEnum
+                    .justicesLegalAdviser && (caseData.getJusticeLegalAdviserFullName() == null || caseData
+                    .getJusticeLegalAdviserFullName().isBlank())) {
+                List<String> errorList = new ArrayList<>();
+                errorList.add("Full name of Justices' Legal Advisor is mandatory, when the Judge's title is selected as Justices' Legal Adviser");
+                return AboutToStartOrSubmitCallbackResponse.builder().errors(errorList).build();
+            }
+            List<String> errorList = new ArrayList<>();
+            if (caseData.getDateOrderMade() == null || caseData.getDateOrderMade().toString().isBlank()) {
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/ManageOrdersController.java:155]: This condition should be refactored

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 6]
```diff
@@ -1414,8 +1431,23 @@ public CallbackResponse handleSelectedOrder(CallbackRequest callbackRequest, Str
...
             DynamicList hearingsDynamicList = manageOrderService.populateHearingsDropdown(authorisation, caseData);
-            manageOrders = manageOrders.toBuilder().hearingsType(hearingsDynamicList).build();
-            caseData = caseData.toBuilder().manageOrders(manageOrders).build();
+            if (manageOrders.getC21OrderOptions() != null) {
+                manageOrders = manageOrders
+                        .toBuilder()
+                        .typeOfC21Order(manageOrders.getC21OrderOptions() != null ? BOLD_BEGIN + manageOrders
+                                .getC21OrderOptions()
+                                .getDisplayedValue() + BOLD_END : "")
+                        .isTheOrderByConsent(Yes)
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:1440]: Why is it hardcoded to Yes?

**Review by shashisk97 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java [hunk 7]
```diff
@@ -226,6 +240,17 @@ public AboutToStartOrSubmitCallbackResponse populateCommonFields(
                 CaseData.class
             );
             Map<String, Object> response = draftAnOrderService.populateCommonDraftOrderFields(authorisation, caseData);
+            CreateSelectOrderOptionsEnum selectedOrder = (CreateSelectOrderOptionsEnum) response.get("orderType");
+            List<String> errorList = new ArrayList<>();
+            String dateOrderMade = response.get("dateOrderMade") != null
+                    ? response.get("dateOrderMade").toString() : " ";
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java:246]: This needs discussion not sure why this condition is added.

Comment (JamesCutts00) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java]: Have removed this

**Pull request closed without merging**